File: 백준/Bronze/1546. 평균/평균.py
def sort(arr):
  for i in range(len(arr) - 1):
    for j in range(i + 1, len(arr)):
      if (arr[i] > arr[j]):
        arr[i], arr[j] = arr[j], arr[i]

  return arr

def getNewaver(score, max_score):
  newAver = score / max_score * 100

  return newAver

a = int(input())

arr = input().split()
score = []
for item in arr:
  score.append(float(item))
sorted_score = sort(score)

sum = 0
for i in range(len(sorted_score)):
  sum += getNewaver(sorted_score[i], sorted_score[len(sorted_score) - 1])

print(sum / a)

File: 백준/Bronze/1546. 평균/평균.c
#include <stdio.h>

double getnewAver(double a, double b){
   double newAver = (a / b) * 100;
   return newAver;
}

void sort(double arr[], int length){
   for(int i = 0; i < length - 1; i++){
      for(int j = i + 1; j < length; j++)
         if(arr[i] > arr[j]){
            double temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
         }
   }
}

int main(void) {
   int a;
   scanf("%d", &a);

   double arr[a];

   for(int i = 0; i < a; i++){
      scanf("%lf", &arr[i]);
   }
   
   sort(arr, a);
   
   double sum = 0;
   for(int i = 0; i < a; i++){
      double newAver = getnewAver(arr[i], arr[a - 1]);
      arr[i] = newAver;
      sum += arr[i];
   }

   printf("%lf", sum / a);
  
   return 0;
}